Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -182,17 +182,30 @@ import BraintreeDataCollector
             payPalContextID: payPalContextID,
             payPalInstalled: payPalAppInstalled
         )
-        guard let url, isValidURLAction(url: url) else {
+
+        guard let url, BTPayPalReturnURL.isValidURLAction(url: url, linkType: linkType) else {
             notifyFailure(with: BTPayPalError.invalidURLAction, completion: completion)
             return
         }
         
-        guard let response = responseDictionary(from: url) else {
+        guard !url.absoluteString.contains("cancel") else {
```

Comment:
The `responseDictionary` method was a bit odd because it was both building the dictionary to pass into the post parameters as well as doing a check that the URL did not contain `cancel`. The only part that used the return URL was the `cancel` check so I broke this out to build the dictionary here which made more sense.